opsdroid/tests/test_connector.py

```diff
@@ -0,0 +1,145 @@
+"""Test the opsdroid connector base class."""
+import pytest
+import asyncio
+
+import asynctest.mock as amock
+
+from opsdroid.core import OpsDroid
+from opsdroid.connector import Connector, register_event
+from opsdroid.events import Message, Reaction
+from opsdroid.cli.start import configure_lang
+
+configure_lang({})
+
+
+@pytest.fixture(scope="session")
+def loop():
+    return asyncio.new_event_loop()
+
+
+def test_init():
+    config = {"example_item": "test"}
+    connector = Connector(config, opsdroid=OpsDroid())
+    assert connector.default_target is None
+    assert connector.name == ""
+    assert connector.config["example_item"] == "test"
+
+
+def test_property():
+    opsdroid = amock.CoroutineMock()
+    connector = Connector({"name": "shell"}, opsdroid=opsdroid)
+    assert connector.configuration.get("name") == "shell"
+
+
+def test_connect(loop):
+    connector = Connector({}, opsdroid=OpsDroid())
+    with pytest.raises(NotImplementedError):
+        loop.run_until_complete(connector.connect())
+
+
+def test_listen(loop):
+    connector = Connector({}, opsdroid=OpsDroid())
```

Comment:
Consider creating a connector fixture to remove code duplication

Reply:
sure.

Reply:
This could be a good idea since we are indeed using connector = Connector... although the only issue I see is that we will have to pass a config to the connector sometimes. 

If we decide to go with the fixture route, we probably will need to update all the other pyttests that test a connector - which is out of scope of this PR

Reply:
I created a fixture for the ones, that does not need config, like this:
```
@pytest.fixture(scope="session")
def no_config_connector():
    return Connector({}, opsdroid=opsdroid)
```
Maybe it is a bit clunky. Other thing could be a higher-order fixture, for which you can pass a config, and it gives you the connector instance.

Reply:
You need to create one fixture with no config by default but the possibility to pass a config as a parameter. This can be achieved writing fixture as a decorator.

Reply:
I changed the fixture. Can you check if you like it, or if a change has to be made.